homeassistant/components/mqtt/fan.py

```diff
@@ -424,18 +429,17 @@ def percentage_received(msg: ReceiveMessage) -> None:
                 )
                 return
             self._attr_percentage = percentage
-            get_mqtt_data(self.hass).state_write_requests.write_state_request(self)
 
         add_subscribe_topic(CONF_PERCENTAGE_STATE_TOPIC, percentage_received)
 
         @callback
         @log_messages(self.hass, self.entity_id)
+        @write_state_on_attr_change(self, {"_attr_preset_mode"})
         def preset_mode_received(msg: ReceiveMessage) -> None:
             """Handle new received MQTT message for preset mode."""
             preset_mode = str(self._value_templates[ATTR_PRESET_MODE](msg.payload))
             if preset_mode == self._payload["PRESET_MODE_RESET"]:
                 self._attr_preset_mode = None
-                self.async_write_ha_state()
```

Comment:
Is this a change in behavior?

Reply:
The wrapper will see `_attr_preset_mode` has changed and call `get_mqtt_data(self.hass).state_write_requests.write_state_request(self)` but previously it would call `self.async_write_ha_state()` directly

Reply:
It looks like all the other platforms call `get_mqtt_data(self.hass).state_write_requests.write_state_request(self)` so this difference doesn't seem intentional

Reply:
Right get_mqtt_data(self.hass).state_write_requests.write_state_request(self) only triggers calling self.async_write_ha_state() once after all subscriptions are processed.